### ISSUE TITLE
Overwriting input image with output image ?

### ISSUE THREAD

**Issue by Adeel-Intizar** [gradio-app/gradio#790]
Hi, how can we show the output image at the place of input image, this means after model inference overwriting the resulted image (OUTPUT) with the input image instead of showing in parallel ?

Thanks

**Comment by abidlabs** [comment]
This is a good suggestion and I think it would allow for to some cool demos. @pngwn @FarukOzderim can we design Blocks so that if the same component is provided as the input and output, the result of the model inference would overwrite the input?  E.g.

```python
im = components.Image()
btn.click(fn, im, im)  # the output of fn overwrites the image used for the input
```

I think we could also build a nice higher-level abstraction `UnifiedInterface` which is designed for these use cases.

**Comment by FarukOzderim** [comment]
I think this will be possible when components are merged into one.

**Comment by pngwn** [comment]
Yeah, i think this should just work but don't quote me on that. Definitely an interesting usecase and something we will want to capture in our tests.

**Comment by FarukOzderim** [comment]
Added it to test workflows!

**Comment by abidlabs** [comment]
This is now fixed in the `blocks-dev` branch!
